app/src/main/kotlin/edu/stanford/bdh/engagehf/bluetooth/component/VitalDisplay.kt

```diff
@@ -0,0 +1,169 @@
+package edu.stanford.bdh.engagehf.bluetooth.component
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.CardDefaults
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.ElevatedCard
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.tooling.preview.PreviewParameterProvider
+import androidx.compose.ui.unit.dp
+import edu.stanford.bdh.engagehf.bluetooth.component.VitalDisplayDataFactory.createVitalDisplayState
+import edu.stanford.bdh.engagehf.bluetooth.data.models.VitalDisplayData
+import edu.stanford.spezi.core.design.theme.Colors
+import edu.stanford.spezi.core.design.theme.Spacings
+import edu.stanford.spezi.core.design.theme.TextStyles
+import edu.stanford.spezi.core.utils.extensions.testIdentifier
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Locale
+
+@Composable
+fun VitalDisplay(
+    modifier: Modifier = Modifier,
+    vitalDisplayUiState: VitalDisplayData,
+) {
+    ElevatedCard(
+        elevation = CardDefaults.elevatedCardElevation(
+            defaultElevation = 4.dp,
+        ),
+        modifier = modifier
+            .padding(Spacings.medium)
+            .height(130.dp)
+            .fillMaxWidth()
+            .testIdentifier(WeightDisplayTestIdentifier.ROOT)
+    ) {
+        Column(
+            modifier = Modifier.padding(Spacings.medium)
+        ) {
+            Text(
+                text = vitalDisplayUiState.title,
+                style = TextStyles.titleMedium,
+                color = Colors.onBackground,
+                modifier = Modifier.testIdentifier(WeightDisplayTestIdentifier.TITLE),
+            )
+            when (vitalDisplayUiState.status) {
+                OperationStatus.SUCCESS -> {
+                    Text(
+                        text = "${vitalDisplayUiState.value} ${vitalDisplayUiState.unit}",
+                        style = MaterialTheme.typography.headlineMedium,
+                        color = Colors.primary,
+                        modifier = Modifier
+                            .padding(vertical = Spacings.small)
+                            .testIdentifier(WeightDisplayTestIdentifier.VALUE),
+                    )
+                    Text(
+                        text = "Measured on: ${vitalDisplayUiState.date}",
```

Comment:
As mentioned above, this text should be constructed in the mapper. Keep also in mind string resources so that we can handle localization as well